packages/airnode-utilities/src/caching/index.ts

```diff
@@ -0,0 +1,252 @@
+ * This is the functional equivalent of "commit" from SQL.
+ */
+const syncFsSync = () => {
+  try {
+    // TODO find a way to mock out spawn...
+    if (!isJest()) {
+      spawn(`sync`, [`-f`, CACHE_BASE_PATH]);
+    }
+  } catch (e) {
+    logger.error(`Unable to sync cache fs`);
+    logger.error((e as Error).stack!);
+  }
+};
+
+/**
+ * Syncs the temporary filesystem asynchronously.
+ * Lambda will "freeze" and "thaw" the execution context and with it any async promises in progress. We create and
+ * forget this promise here, but it will resolve at some point, either in this invocation or a later one.
+ */
+const syncFsASync = () => {
+  try {
+    new Promise(() => {
+      spawn(`sync`, [`-f`, CACHE_BASE_PATH]);
+    });
+  } catch (e) {
+    logger.error(`Unable to sync cache fs`);
+    logger.error((e as Error).stack!);
+  }
+};
+
+/**
+ * Initialise the path for the fs cache
+ */
+const initPath = () => {
+  try {
+    if (!existsSync(CACHE_BASE_PATH)) {
+      mkdirSync(CACHE_BASE_PATH);
+      syncFsSync();
+    }
+  } catch (e) {
+    logger.error(`Unable to init fs cache`);
+    logger.error((e as Error).stack!);
+  }
+};
+
+/**
+ * Get all keys from the cache that start with startingKey.
+ *
+ * @param startingKey should take the form of `blockedRequestId-` and the full key could be `blockedRequestId-0x0000000...`
+ */
+const getKeys = (startingKey?: string): string[] => {
+  try {
+    const keys = readdirSync(CACHE_BASE_PATH);
+    if (startingKey) {
+      return keys.filter((key) => key.indexOf(startingKey) === 0);
+    }
+    return keys;
+  } catch (e) {
+    logger.error(`Unable to retrieve keys from fs cache`);
+    logger.error((e as Error).stack!);
+  }
+
+  return [];
+};
+
+/**
+ * Get the value stored for a key (aka the contents of the file with the key as name).
+ *
+ * @param key the key whose contents should be returned
+ */
+const getValueForKey = (key: string): any | undefined => {
+  try {
+    if (!existsSync(join(CACHE_BASE_PATH, key))) {
+      return undefined;
+    }
+
+    return JSON.parse(readFileSync(join(CACHE_BASE_PATH, key)).toString());
+  } catch (e) {
+    logger.error(`Unable to read key data from fs cache`);
+    logger.error((e as Error).stack!);
+  }
+
+  return undefined;
+};
+
+/**
+ * Remove a key, this is discouraged and should only be used by sweep operations (garbage collection).
+ *
+ * @param key the key to remove
+ */
+const removeKey = (key: string) => {
+  try {
+    // Force will not throw an error if the file doesn't exist
+    rmSync(join(CACHE_BASE_PATH, key), { force: true });
+  } catch (e) {
+    logger.error(`Unable to remove key from fs cache`);
+    logger.error((e as Error).stack!);
+  }
+};
+
+/**
+ * Add a cache entry
+ *
+ * @param key the name of the key, ideally in the form of `something-{uniqueId}`, eg. `requestIdBlocked-0x0000...`
+ * @param data the data to store for the key, arbitrary
+ * @param force force overwriting of an existing key (discouraged)
```

Comment:
Mhh, is this really discouraged? Won't this be a relatively common operation (refresh the data returned by the API)?

Reply:
One of the big use cases of this is to cache an API response for a unique `requestId`, in these cases you won't overwrite the API response because if the key exists you'd want to return the value for that key to the requester.

This is for when API operations are non-idempotent and where a request doesn't get honoured on the first, second, etc. cycle.

It should be fine @ force, I was being paranoid!